src/ProtectedProxies/MinimalProtectedProxyFactory.sol

```diff
@@ -25,14 +26,13 @@ contract MinimalProtectedProxyFactory is SphereXConfiguration {
 
     event DeployedMinimalProtectedProxy(address proxy_address, address implementation);
 
-
     modifier onlyAllowedDeployer() {
         require(_allowedDeployer != address(0), "Must define Allowed deployer");
         require(msg.sender == _allowedDeployer, "Only Allowed Deployer");
         _;
     }
 
-    function initializeAllowedDeployer(address allowedDeployer) onlySphereXAdmin external {
+    function initializeAllowedDeployer(address allowedDeployer) external onlySphereXAdmin {
         require(_allowedDeployer == address(0), "Allowed deployer already exists");
         _allowedDeployer = allowedDeployer;
```

Comment:
Why is this initializeable and not immutable?

Reply:
because of dependency loop